<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Anthony Hung" />

<meta name="date" content="2021-01-19" />

<title>Pre-processing of raw 10x files into count matrices and demultiplexing</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">invitroOA_pilot_repository</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Pre-processing of raw 10x files into count matrices and demultiplexing</h1>
<h4 class="author">Anthony Hung</h4>
<h4 class="date">2021-01-19</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-01-21
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>invitroOA_pilot_repository/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210119code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210119)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210119code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210119)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomanthonyhunginvitroOApilotrepositorytree58e93fde90f2d936b7dff46f3905f1d0b713b9e8targetblank58e93fda"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/anthonyhung/invitroOA_pilot_repository/tree/58e93fde90f2d936b7dff46f3905f1d0b713b9e8" target="_blank">58e93fd</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomanthonyhunginvitroOApilotrepositorytree58e93fde90f2d936b7dff46f3905f1d0b713b9e8targetblank58e93fda" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/anthonyhung/invitroOA_pilot_repository/tree/58e93fde90f2d936b7dff46f3905f1d0b713b9e8" target="_blank">58e93fd</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/bulkRNA_preprocessing/.snakemake/conda-archive/
    Ignored:    code/bulkRNA_preprocessing/.snakemake/conda/
    Ignored:    code/bulkRNA_preprocessing/.snakemake/locks/
    Ignored:    code/bulkRNA_preprocessing/.snakemake/shadow/
    Ignored:    code/bulkRNA_preprocessing/.snakemake/singularity/
    Ignored:    code/bulkRNA_preprocessing/.snakemake/tmp.3ekfs3n5/
    Ignored:    code/bulkRNA_preprocessing/fastq/
    Ignored:    code/bulkRNA_preprocessing/out/
    Ignored:    code/single_cell_preprocessing/.snakemake/conda-archive/
    Ignored:    code/single_cell_preprocessing/.snakemake/conda/
    Ignored:    code/single_cell_preprocessing/.snakemake/locks/
    Ignored:    code/single_cell_preprocessing/.snakemake/shadow/
    Ignored:    code/single_cell_preprocessing/.snakemake/singularity/
    Ignored:    code/single_cell_preprocessing/YG-AH-2S-ANT-1_S1_L008/
    Ignored:    code/single_cell_preprocessing/YG-AH-2S-ANT-2_S2_L008/
    Ignored:    code/single_cell_preprocessing/demuxlet/.DS_Store
    Ignored:    code/single_cell_preprocessing/fastq/
    Ignored:    data/external_scRNA/Chou_et_al2020/
    Ignored:    data/external_scRNA/Jietal2018/
    Ignored:    data/external_scRNA/Wuetal2021/
    Ignored:    data/external_scRNA/merged_external_scRNA.rds
    Ignored:    data/poweranalysis/alasoo_et_al/
    Ignored:    output/GO_terms_enriched.csv
    Ignored:    output/topicModel_k=7.rds
    Ignored:    output/voom_results.rds

Unstaged changes:
    Modified:   .gitignore

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/preProcess_scRNA.Rmd</code>) and HTML (<code>docs/preProcess_scRNA.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/58e93fde90f2d936b7dff46f3905f1d0b713b9e8/analysis/preProcess_scRNA.Rmd" target="_blank">58e93fd</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
<td>
forgo the grid
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/anthonyhung/invitroOA_pilot_repository/58e93fde90f2d936b7dff46f3905f1d0b713b9e8/docs/preProcess_scRNA.html" target="_blank">58e93fd</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
<td>
forgo the grid
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/89d8ddd8f60efeed6a2ee39010b89c66d4cadb48/analysis/preProcess_scRNA.Rmd" target="_blank">89d8ddd</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
<td>
add three rows
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/anthonyhung/invitroOA_pilot_repository/89d8ddd8f60efeed6a2ee39010b89c66d4cadb48/docs/preProcess_scRNA.html" target="_blank">89d8ddd</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
<td>
add three rows
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/32e5716c8d339917866ab21325353d2cd2d3cc77/analysis/preProcess_scRNA.Rmd" target="_blank">32e5716</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
<td>
add gridextra
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/78cfbcd3204249a8673228d684ae77444c63266b/analysis/preProcess_scRNA.Rmd" target="_blank">78cfbcd</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
<td>
finish paring down files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/e0e89e18e26a5e4312f081c98eddb2d0f9577653/analysis/preProcess_scRNA.Rmd" target="_blank">e0e89e1</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
<td>
download annotations and finish preprocessing of bulkRNA
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/99c70b8e43338b294c4aa7877ea7eccc732be2ac/analysis/preProcess_scRNA.Rmd" target="_blank">99c70b8</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-20
</td>
<td>
update external data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/anthonyhung/invitroOA_pilot_repository/99c70b8e43338b294c4aa7877ea7eccc732be2ac/docs/preProcess_scRNA.html" target="_blank">99c70b8</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-20
</td>
<td>
update external data
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/446bf4b363cf39fc8bfa6f4868ba662b6793051a/analysis/preProcess_scRNA.Rmd" target="_blank">446bf4b</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-20
</td>
<td>
add to gitignore
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/1ec66ea82cb6f4c6fc39b238c58fca311f44cf64/analysis/preProcess_scRNA.Rmd" target="_blank">1ec66ea</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-20
</td>
<td>
add information desribing preprocessing scran
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/28f57faed7255644b4858290aed125d5466312ea/analysis/preProcess_scRNA.Rmd" target="_blank">28f57fa</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-19
</td>
<td>
Add files for analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This page will walk through the steps to go from the raw 10x sequencing fastq files to a count matrix and demuxlet assignment of droplets to individuals. This involves running a Snakemake pipeline located in the code directory and some code in R. The end product are seurat objects containing the raw count matrices and assignment to one of the three individuals and two treatment groups (strain or unstrain) for each of the two 10x GEM wells used in the sequencing experiment.</p>
</div>
<div id="run-the-snakemake-pipeline-after-downloading-necessary-files-and-creating-the-conda-environment-see-readme.txt" class="section level1">
<h1>Run the Snakemake Pipeline after downloading necessary files and creating the conda environment (see README.txt)</h1>
<ol style="list-style-type: decimal">
<li><p>Move the fastq files from all samples (these are output files from 10x runs, containing both forward and reverse sequences) into the folder code/single_cell_preprocessing/fastq/. Undetermined data files are not required.</p></li>
<li><p>Modify the code/single_cell_preprocessing/Pipeline/cluster_solo.json file to correspond to the computing cluster you are working with.</p></li>
<li><p>Unzip the whitelist file in the code/single_cell_preprocessing/ directory.</p></li>
<li><p>Place the human.YRI.cellranger.exons.vcf file into the code/single_cell_preprocessing/ directory.</p></li>
<li><p>Install the conda working by running “conda env create –file environment.yaml”</p></li>
<li><p>Run “source activate dropseq2” to activate the conda environment.</p></li>
<li><p>Run the file “submit.sh”.</p></li>
</ol>
</div>
<div id="load-files-into-r" class="section level1">
<h1>Load files into R</h1>
<p>After running the pipeline, two directories will be created corresponding to the two 10x GEM well involved in the sequencing experiment (“YG-AH-2S-ANT-1_S1_L008” and “YG-AH-2S-ANT-2_S2_L008”). These directories contain the demuxlet and STAR SOLO outputs for each 10x GEM well</p>
<p>Details about the individualXtreatment status of the samples that were pooled for each of the two GEM wells: ANT1: (NA19160 unstrained; NA18856 unstrained; NA18855 strained) ANT2: (NA19160 strained; NA18855 unstrained)</p>
<pre class="r"><code>library(data.table)
library(Matrix)
library(Seurat)
library(readr)
library(stringr)
library(plyr)
library(dplyr)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:plyr&#39;:

    arrange, count, desc, failwith, id, mutate, rename, summarise,
    summarize</code></pre>
<pre><code>The following objects are masked from &#39;package:data.table&#39;:

    between, first, last</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>## link to directories containing data files (count matrices)
proj_dir &lt;- &quot;code/single_cell_preprocessing/&quot;
ANT1_dir &lt;- paste0(proj_dir, &quot;YG-AH-2S-ANT-1_S1_L008/&quot;)
ANT2_dir &lt;- paste0(proj_dir, &quot;YG-AH-2S-ANT-2_S2_L008/&quot;)

#read in data

##Gene Output from STARSOLO
#ANT1
demuxlet1 &lt;- fread(paste0(ANT1_dir, &quot;demuxlet.best&quot;, sep = &quot;&quot;))
count_data1 &lt;- readMM(paste0(ANT1_dir, &quot;Gene/filtered/matrix.mtx&quot;))
genes1 &lt;- read_tsv(paste0(ANT1_dir, &quot;Gene/filtered/genes.tsv&quot;), col_names = F)</code></pre>
<pre><code>Parsed with column specification:
cols(
  X1 = col_character(),
  X2 = col_character()
)</code></pre>
<pre class="r"><code>barcodes1 &lt;- as.data.frame(read_tsv(paste0(ANT1_dir, &quot;Gene/filtered/barcodes.tsv&quot;), col_names = F))</code></pre>
<pre><code>Parsed with column specification:
cols(
  X1 = col_character()
)</code></pre>
<pre class="r"><code>#ANT2
demuxlet2 &lt;- fread(paste0(ANT2_dir, &quot;demuxlet.best&quot;, sep = &quot;&quot;))
count_data2 &lt;- readMM(paste0(ANT2_dir, &quot;Gene/filtered/matrix.mtx&quot;))
genes2 &lt;- read_tsv(paste0(ANT2_dir, &quot;Gene/filtered/genes.tsv&quot;), col_names = F)</code></pre>
<pre><code>Parsed with column specification:
cols(
  X1 = col_character(),
  X2 = col_character()
)</code></pre>
<pre class="r"><code>barcodes2 &lt;- as.data.frame(read_tsv(paste0(ANT2_dir, &quot;Gene/filtered/barcodes.tsv&quot;), col_names = F))</code></pre>
<pre><code>Parsed with column specification:
cols(
  X1 = col_character()
)</code></pre>
</div>
<div id="based-on-the-demuxlet-output-assign-label-for-barcodes-based-on-best-output-and-filter-for-sng--barcodes" class="section level1">
<h1>Based on the demuxlet output, assign label for barcodes based on “BEST” output and filter for “SNG-” barcodes</h1>
<pre class="r"><code>#this function returns a dataframe with two columns, one corresponding to the barcodes and one corresponding to the label given by demuxlet
return_singlet_label &lt;- function(barcodes, demuxlet.out){
  labels &lt;- demuxlet.out$BEST[match(unlist(barcodes), demuxlet.out$BARCODE)]
  return(cbind(barcodes, labels))
}

barcodes1_labeled &lt;- return_singlet_label(barcodes1, demuxlet1)
barcodes2_labeled &lt;- return_singlet_label(barcodes2, demuxlet2)

#table of singlets/multiplets in the filtered data based on demuxlet
table(barcodes1_labeled$labels)</code></pre>
<pre><code>
DBL-NA18855-NA18856-0.500 DBL-NA19160-NA18855-0.500 
                       13                         6 
DBL-NA19160-NA18856-0.500               SNG-NA18855 
                        7                       411 
              SNG-NA18856               SNG-NA19160 
                      452                       260 </code></pre>
<pre class="r"><code>table(barcodes2_labeled$labels)</code></pre>
<pre><code>
DBL-NA18855-NA18856-0.500 DBL-NA18855-NA19160-0.500 
                       20                      2719 
DBL-NA18856-NA18855-0.500 DBL-NA19160-NA18855-0.500 
                        4                      5650 
DBL-NA19160-NA18856-0.500               SNG-NA18855 
                       22                      4990 
              SNG-NA19160 
                     1506 </code></pre>
<pre class="r"><code>## filter for droplets in the count data that are singlets (remove multiplets)
#ANT1
demuxlet_single1 &lt;- demuxlet1 %&gt;%
     dplyr::filter(grepl(&quot;SNG-&quot;, BEST))
singlets_index1 &lt;- unlist(lapply(barcodes1_labeled$X1,&quot;%in%&quot;, table = demuxlet_single1$BARCODE), use.names = F) #get index of barcodes that are singlets
barcodes_singlets1 &lt;- barcodes1_labeled[singlets_index1,] #use index to subset matrix + barcode names
count_data_singlets1 &lt;- count_data1[,singlets_index1]

#ANT2
demuxlet_single2 &lt;- demuxlet2 %&gt;%
     dplyr::filter(grepl(&quot;SNG-&quot;, BEST))
singlets_index2 &lt;- unlist(lapply(barcodes2_labeled$X1,&quot;%in%&quot;, table = demuxlet_single2$BARCODE), use.names = F) #get index of barcodes that are singlets
barcodes_singlets2 &lt;- barcodes2_labeled[singlets_index2,] #use index to subset matrix + barcode names
count_data_singlets2 &lt;- count_data2[,singlets_index2]</code></pre>
</div>
<div id="create-seurat-object-for-each-dataset-for-singlet-barcodes-and-add-metadata-in-the-form-of-singlet-identity-for-each-barcode." class="section level1">
<h1>Create Seurat object for each dataset (for singlet barcodes) and add metadata in the form of singlet identity for each barcode.</h1>
<pre class="r"><code>#Change labels to reflect strain/unstrain based on prior knowledge of which strainXindividual combinations went into each pool

strainIndlabels1 &lt;- revalue(barcodes_singlets1$labels,
                            c(&quot;SNG-NA18856&quot;= &quot;NA18856_Unstrain&quot;,
                              &quot;SNG-NA18855&quot; = &quot;NA18855_Strain&quot;,
                              &quot;SNG-NA19160&quot; = &quot;NA19160_Unstrain&quot;))

strainIndlabels2 &lt;- revalue(barcodes_singlets2$labels,
                            c(&quot;SNG-NA18855&quot; = &quot;NA18855_Unstrain&quot;,
                              &quot;SNG-NA19160&quot; = &quot;NA19160_Strain&quot;))


rownames(count_data_singlets1) &lt;- genes1$X2
colnames(count_data_singlets1) &lt;- barcodes_singlets1$X1

ANT1_seurat &lt;- CreateSeuratObject(counts = count_data_singlets1, project = &quot;ANT1&quot;) %&gt;%
  AddMetaData(strainIndlabels1, col.name = &quot;labels&quot;)</code></pre>
<pre><code>Warning: Non-unique features (rownames) present in the input matrix, making
unique</code></pre>
<pre class="r"><code>rownames(count_data_singlets2) &lt;- genes2$X2
colnames(count_data_singlets2) &lt;- barcodes_singlets2$X1

ANT2_seurat &lt;- CreateSeuratObject(counts = count_data_singlets2, project = &quot;ANT2&quot;) %&gt;%
  AddMetaData(strainIndlabels2, col.name = &quot;labels&quot;)</code></pre>
<pre><code>Warning: Non-unique features (rownames) present in the input matrix, making
unique</code></pre>
</div>
<div id="merge-the-two-seurat-objects-without-any-data-integration-and-filter-out-unwanted-cells-based-on-qc-metrics" class="section level1">
<h1>Merge the two seurat objects (without any data integration) and filter out unwanted cells based on QC metrics</h1>
<p>This merged dataset is used to fit the topic model in a later file. Based on the clustering results, which show that the cells from the same individual from different 10x GEM well overlap, there does not seem to be a large contribution of technical effects from 10x GEM well on gene expression.</p>
<pre class="r"><code>ANT1.2 &lt;- merge(x = ANT1_seurat,
                   y = ANT2_seurat,
                   add.cell.ids = c(&quot;ANT1&quot;, &quot;ANT2&quot;),
                   merge.data = F,
                   project = &quot;OAStrain&quot;)

#compute the percentage of reads coming from mitochondrial genes for each droplet
ANT1.2[[&quot;percent.mt&quot;]] &lt;- PercentageFeatureSet(ANT1.2, pattern = &quot;^MT-&quot;)

#visualize QC metrics as violin plot
VlnPlot(ANT1.2, features = c(&quot;nFeature_RNA&quot;, &quot;nCount_RNA&quot;, &quot;percent.mt&quot;), ncol = 3)</code></pre>
<p><img src="figure/preProcess_scRNA.Rmd/merge-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-merge-1">
Past versions of merge-1.png
</button>
</p>
<div id="fig-merge-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/89d8ddd8f60efeed6a2ee39010b89c66d4cadb48/docs/figure/preProcess_scRNA.Rmd/merge-1.png" target="_blank">89d8ddd</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(ANT1.2, features = c(&quot;nFeature_RNA&quot;, &quot;nCount_RNA&quot;, &quot;percent.mt&quot;), ncol = 3, group.by = &quot;labels&quot;)</code></pre>
<p><img src="figure/preProcess_scRNA.Rmd/merge-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-merge-2">
Past versions of merge-2.png
</button>
</p>
<div id="fig-merge-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/89d8ddd8f60efeed6a2ee39010b89c66d4cadb48/docs/figure/preProcess_scRNA.Rmd/merge-2.png" target="_blank">89d8ddd</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># FeatureScatter is typically used to visualize feature-feature relationships, but can be used
# for anything calculated by the object, i.e. columns in object metadata, PC scores etc.
plot1 &lt;- FeatureScatter(ANT1.2, feature1 = &quot;nCount_RNA&quot;, feature2 = &quot;percent.mt&quot;)
plot2 &lt;- FeatureScatter(ANT1.2, feature1 = &quot;nCount_RNA&quot;, feature2 = &quot;nFeature_RNA&quot;)
CombinePlots(plots = list(plot1, plot2))</code></pre>
<pre><code>Warning: CombinePlots is being deprecated. Plots should now be combined
using the patchwork system.</code></pre>
<p><img src="figure/preProcess_scRNA.Rmd/merge-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-merge-3">
Past versions of merge-3.png
</button>
</p>
<div id="fig-merge-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/89d8ddd8f60efeed6a2ee39010b89c66d4cadb48/docs/figure/preProcess_scRNA.Rmd/merge-3.png" target="_blank">89d8ddd</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#Filter barcodes based on nFeatures and %MT 
ANT1.2 &lt;- subset(ANT1.2, subset = nFeature_RNA &gt; 2000 &amp; percent.mt &lt; 10)
table(ANT1.2$labels)</code></pre>
<pre><code>
  NA18855_Strain NA18855_Unstrain NA18856_Unstrain   NA19160_Strain 
             348              586              395              264 
NA19160_Unstrain 
             222 </code></pre>
<pre class="r"><code>#Look at QC metrics after filtering
VlnPlot(ANT1.2, features = c(&quot;nFeature_RNA&quot;, &quot;nCount_RNA&quot;, &quot;percent.mt&quot;), ncol = 3, group.by = &quot;labels&quot;)</code></pre>
<p><img src="figure/preProcess_scRNA.Rmd/merge-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-merge-4">
Past versions of merge-4.png
</button>
</p>
<div id="fig-merge-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/89d8ddd8f60efeed6a2ee39010b89c66d4cadb48/docs/figure/preProcess_scRNA.Rmd/merge-4.png" target="_blank">89d8ddd</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dim(ANT1.2)</code></pre>
<pre><code>[1] 33538  1815</code></pre>
<pre class="r"><code># Look at how the samples cluster after merging without any integration methods applied
#normalize
ANT1.2 &lt;- NormalizeData(ANT1.2, normalization.method = &quot;LogNormalize&quot;, scale.factor = 10000)
#find variable features and run a PCA
ANT1.2 &lt;- FindVariableFeatures(ANT1.2, selection.method = &quot;vst&quot;, nfeatures = 2000)
ANT1.2 &lt;- ScaleData(ANT1.2, verbose = FALSE)
nPC &lt;- 100
ANT1.2 &lt;- RunPCA(ANT1.2, features = VariableFeatures(object = ANT1.2), npcs = nPC)</code></pre>
<pre><code>PC_ 1 
Positive:  VIM, TIMP3, INHBA, TPM1, MAP3K7CL, SPINT2, PDLIM3, SOX4, C12orf75, TUBA1A 
       CCND2, CPE, RDH10, LZTS1, PMEPA1, LMO7, CDKN2B, TCAF1, MARCKSL1, UCHL1 
       GSTO1, XRCC4, IGFL2, FN1, FSCN1, RAI14, STMN1, CTGF, JPT1, MYH9 
Negative:  IFITM3, SAA1, SPON2, PDPN, TMEM141, PTGDS, DPP7, OTULINL, CDKN1C, TSPO 
       PRLR, DCN, LUM, FKBP5, PNMT, GGT5, MMP23B, NUPR1, TMEM88, PLIN2 
       SERPINF1, IQGAP2, ADAMTS2, SMOC2, APOD, GMPR, FAM213A, FMO5, TXNDC16, MME 
PC_ 2 
Positive:  RPS4Y1, SLC3A2, EIF1AY, FTH1, SERPINE1, PTGR1, DDIT3, EIF1B, FAM129A, SQSTM1 
       DDX3Y, SPON2, MT1E, MGST1, CEBPB, BTG1, CLIC1, TRIB3, KCNG1, EZR 
       DUSP1, CYSTM1, PRDX6, TSPO, IFRD1, IMPA2, MAGEH1, NFIL3, YWHAQ, BLVRB 
Negative:  CHCHD2, MDK, MFAP4, PLAC9, OGN, COL15A1, B2M, SLC40A1, COL6A3, MEST 
       SPON1, FIBIN, HLA-A, MXRA5, ACTA2, RIMS2, IGFBP3, KCTD12, PITX1, PRSS35 
       CEMIP, GLIPR1, COL6A1, ADAM12, CCND2, SBF2-AS1, LURAP1L, FST, GNG11, CHN1 
PC_ 3 
Positive:  OGFRL1, SMOC2, AQP1, TMEM88, GPC3, SELENOP, PROCR, ODAPH, CLTB, PRDX6 
       PLAC9, MEST, IGFBP7, PLA2G16, FLT1, SLIT3, ATRNL1, IGFBP2, TSPAN15, NEDD9 
       PDZRN4, TMEM176B, HAPLN4, RGS4, SRPX, NSG1, COL4A4, AC078850.1, TMEM176A, TPD52L1 
Negative:  MGST1, TGFBI, DDIT4, TRIB3, ZFAS1, FN1, SLC7A11, LGALS1, FTH1, SLFN5 
       AP002884.1, TXNRD1, LIMS1, GCLM, COL6A3, C1R, CRABP2, TSC22D3, PHGDH, HLA-B 
       ASNS, SLC12A8, DDIT3, NFIL3, NQO1, PMAIP1, TCEA1, GPR1, PCK2, AC040170.1 
PC_ 4 
Positive:  IL6ST, TIMP1, COL6A2, COL6A1, HOPX, FBLN2, NR2F1, LTBP1, COL15A1, F3 
       SLC22A3, MME, MFAP4, MYL10, PDGFRA, SPON1, FLT1, TGM2, GAS7, TIMP3 
       CLMP, EGFL6, GDF6, EGFLAM, HGF, VTN, TSPAN9, FGF7, LTBP2, COL11A1 
Negative:  NRG1, MFAP5, ANOS1, ODAPH, PROCR, HTRA1, NKAIN4, AL356056.1, ANXA3, SPINT1 
       ODC1, PRSS23, SMOC2, ANGPTL7, PDZRN4, TXNRD2, TUBA1A, ENAH, CDH6, NTRK2 
       CLU, CDKN2A, DYNC1I1, SFRP1, PLAC9, PPME1, GADD45A, NCAM1, PLXDC1, GATM 
PC_ 5 
Positive:  CDK1, PCLAF, CENPM, UBE2C, RRM2, TOP2A, NUSAP1, PTTG1, TYMS, PBK 
       BIRC5, CENPF, HMMR, SPC25, DEPDC1, CEP55, TUBA1B, FOXM1, SHCBP1, HMGB2 
       ORC6, NCAPG, GINS2, TROAP, TK1, GTSE1, MAD2L1, TPX2, ANLN, KRT18 
Negative:  FN1, FST, THBS2, WNT2, CXCL14, LOX, FILIP1L, NEAT1, RPRML, RAMP1 
       LUM, CKB, HTRA1, PLXDC1, TSPAN13, CCND2, FAP, COL6A2, IFI16, PLXDC2 
       PSPN, EGFL6, COMP, TDO2, RCN3, HSPB6, RPS4Y1, CD55, GPC4, IGFL3 </code></pre>
<pre class="r"><code>#Run clustering and UMAP
num_PCs &lt;- 50
ANT1.2 &lt;- FindNeighbors(ANT1.2, dims = 1:num_PCs)</code></pre>
<pre><code>Computing nearest neighbor graph</code></pre>
<pre><code>Computing SNN</code></pre>
<pre class="r"><code>ANT1.2 &lt;- FindClusters(ANT1.2, resolution = 0.5)</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 1815
Number of edges: 85550

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8146
Number of communities: 7
Elapsed time: 0 seconds</code></pre>
<pre class="r"><code>#Run UMAP
ANT1.2 &lt;- RunUMAP(ANT1.2, dims = 1:num_PCs)</code></pre>
<pre><code>Warning: The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric
To use Python UMAP via reticulate, set umap.method to &#39;umap-learn&#39; and metric to &#39;correlation&#39;
This message will be shown once per session</code></pre>
<pre><code>17:40:41 UMAP embedding parameters a = 0.9922 b = 1.112</code></pre>
<pre><code>17:40:41 Read 1815 rows and found 50 numeric columns</code></pre>
<pre><code>17:40:41 Using Annoy for neighbor search, n_neighbors = 30</code></pre>
<pre><code>17:40:41 Building Annoy index with metric = cosine, n_trees = 50</code></pre>
<pre><code>0%   10   20   30   40   50   60   70   80   90   100%</code></pre>
<pre><code>[----|----|----|----|----|----|----|----|----|----|</code></pre>
<pre><code>**************************************************|
17:40:42 Writing NN index file to temp file /tmp/RtmpC9TKaD/file3919e8491c1d
17:40:42 Searching Annoy index using 1 thread, search_k = 3000
17:40:42 Annoy recall = 100%
17:40:42 Commencing smooth kNN distance calibration using 1 thread
17:40:43 Initializing from normalized Laplacian + noise
17:40:43 Commencing optimization for 500 epochs, with 73932 positive edges
17:40:49 Optimization finished</code></pre>
<pre class="r"><code>p1 &lt;- DimPlot(ANT1.2, reduction = &quot;umap&quot;)
p2 &lt;- DimPlot(ANT1.2, reduction = &quot;umap&quot;, group.by = &quot;orig.ident&quot;)
p3 &lt;- DimPlot(ANT1.2, reduction = &quot;umap&quot;, group.by = &quot;labels&quot;)
p1</code></pre>
<p><img src="figure/preProcess_scRNA.Rmd/merge-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-merge-5">
Past versions of merge-5.png
</button>
</p>
<div id="fig-merge-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/58e93fde90f2d936b7dff46f3905f1d0b713b9e8/docs/figure/preProcess_scRNA.Rmd/merge-5.png" target="_blank">58e93fd</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/89d8ddd8f60efeed6a2ee39010b89c66d4cadb48/docs/figure/preProcess_scRNA.Rmd/merge-5.png" target="_blank">89d8ddd</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>p2</code></pre>
<p><img src="figure/preProcess_scRNA.Rmd/merge-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p3</code></pre>
<p><img src="figure/preProcess_scRNA.Rmd/merge-7.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="save-seurat-objects" class="section level1">
<h1>Save Seurat objects</h1>
<pre class="r"><code>saveRDS(ANT1.2, &quot;data/ANT1_2.rds&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] dplyr_1.0.2       plyr_1.8.6        stringr_1.4.0     readr_1.3.1      
[5] Seurat_3.2.3      Matrix_1.2-18     data.table_1.13.0

loaded via a namespace (and not attached):
  [1] Rtsne_0.15            colorspace_2.0-0      deldir_0.1-23        
  [4] ellipsis_0.3.1        ggridges_0.5.1        rprojroot_2.0.2      
  [7] fs_1.3.1              spatstat.data_1.7-0   farver_2.0.3         
 [10] leiden_0.3.1          listenv_0.8.0         npsurv_0.4-0         
 [13] ggrepel_0.9.0         RSpectra_0.15-0       codetools_0.2-16     
 [16] splines_3.6.1         lsei_1.2-0            knitr_1.23           
 [19] polyclip_1.10-0       jsonlite_1.7.2        workflowr_1.6.2      
 [22] ica_1.0-2             cluster_2.1.0         png_0.1-7            
 [25] uwot_0.1.10           shiny_1.3.2           sctransform_0.3.2    
 [28] compiler_3.6.1        httr_1.4.2            lazyeval_0.2.2       
 [31] later_1.1.0.1         htmltools_0.5.0       tools_3.6.1          
 [34] rsvd_1.0.1            igraph_1.2.4.1        gtable_0.3.0         
 [37] glue_1.4.2            RANN_2.6.1            reshape2_1.4.3       
 [40] rappdirs_0.3.1        Rcpp_1.0.5            spatstat_1.64-1      
 [43] scattermore_0.7       vctrs_0.3.6           gdata_2.18.0         
 [46] nlme_3.1-140          lmtest_0.9-37         xfun_0.8             
 [49] globals_0.12.5        mime_0.9              miniUI_0.1.1.1       
 [52] lifecycle_0.2.0       irlba_2.3.3           gtools_3.8.1         
 [55] goftest_1.2-2         future_1.18.0         MASS_7.3-52          
 [58] zoo_1.8-8             scales_1.1.1          hms_0.5.3            
 [61] promises_1.1.1        spatstat.utils_1.17-0 parallel_3.6.1       
 [64] RColorBrewer_1.1-2    yaml_2.2.1            reticulate_1.16      
 [67] pbapply_1.4-0         gridExtra_2.3         ggplot2_3.3.3        
 [70] rpart_4.1-15          stringi_1.4.6         caTools_1.17.1.2     
 [73] rlang_0.4.10          pkgconfig_2.0.3       matrixStats_0.57.0   
 [76] bitops_1.0-6          evaluate_0.14         lattice_0.20-41      
 [79] ROCR_1.0-7            purrr_0.3.4           tensor_1.5           
 [82] labeling_0.4.2        patchwork_1.1.0       htmlwidgets_1.5.2    
 [85] cowplot_1.1.0         tidyselect_1.1.0      RcppAnnoy_0.0.18     
 [88] magrittr_2.0.1        R6_2.5.0              gplots_3.0.1.1       
 [91] generics_0.0.2        withr_2.3.0           pillar_1.4.7         
 [94] whisker_0.3-2         mgcv_1.8-28           fitdistrplus_1.0-14  
 [97] survival_2.44-1.1     abind_1.4-5           tibble_3.0.4         
[100] future.apply_1.3.0    crayon_1.3.4          KernSmooth_2.23-15   
[103] plotly_4.9.2.1        rmarkdown_1.13        grid_3.6.1           
[106] git2r_0.26.1          digest_0.6.27         xtable_1.8-4         
[109] tidyr_1.1.2           httpuv_1.5.1          munsell_0.5.0        
[112] viridisLite_0.3.0    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
