---
title: "DEanalysis_bulkRNA"
author: "Anthony Hung"
date: "2021-01-19"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

# Introduction

This code takes as input the filtered count data and two factors of unwanted variation fit by RUVs. It then uses limma and voom to perform differential expression analysis between treatment conditions using the data. After we have a set of differentially expressed genes, we look for enrichments amongst DE genes in GO pathways and previously published osteoarthritis datasets.

# Load libraries and data

```{r loading libraries and data, message=F, warning=F}
library("limma")
library("plyr")
library("dplyr")
library("edgeR")
library("tidyr")
library("ashr")
library("ggplot2")
library("RUVSeq")
library("topGO")

#load in filtered count data, RUVs output
filt_counts <- readRDS("data/filtered_counts.rds")
filt_counts <- filt_counts$counts
RUVsOut <- readRDS("data/RUVsOut.rds")

# load gene annotations
gene_anno <- read.delim("data/gene-annotation.txt",
                        sep = "\t")

# load in reordered sample information
sampleinfo <- readRDS("data/Sample.info.RNAseq.reordered.rds")
```

# Specify design matrix for Linear mixed model

```{r design-matrix}
anno <- pData(RUVsOut)
x <- paste0(anno$Individual, anno$treatment)

anno$LibraryPrepBatch <- factor(anno$LibraryPrepBatch, levels = c("1", "2"))
anno$Replicate <- factor(anno$Replicate, levels = c("1", "2", "3"))

design <- model.matrix(~treatment + Individual + W_1 + W_2 + RIN,
                       data = anno)
colnames(design) <- gsub("treatment", "", colnames(design))
colnames(design)
```

```{r fit-model, fig.width=8}
# Model replicate as a random effect
# Recommended to run both voom and duplicateCorrelation twice.
# https://support.bioconductor.org/p/59700/#67620

#TMM Normalization
y <- DGEList(filt_counts)
y <- calcNormFactors(y, method = "TMM")
#Voom for differential expression
v1 <- voom(y, design)
corfit1 <- duplicateCorrelation(v1, design, block = x)
corfit1$consensus.correlation
v2 <- voom(y, design, block = x, correlation = corfit1$consensus.correlation)
corfit2 <- duplicateCorrelation(v2, design, block = x)
corfit2$consensus.correlation
fit <- lmFit(v2, design, block = x,
             correlation = corfit2$consensus.correlation)
fit <- eBayes(fit)

saveRDS(v2, "output/voom_results.rds")
```

## Assess model results

```{r model-results}
get_results <- function(x, number = nrow(x$coefficients), sort.by = "none",
                        ...) {
  # x - object MArrayLM from eBayes output
  # ... - additional arguments passed to topTable
  stopifnot(class(x) == "MArrayLM")
  results <- topTable(x, number = number, sort.by = sort.by, ...)
  return(results)
}

top_treatment <- get_results(fit, coef = "Unstrain", sort.by = "B")
head(top_treatment)

results_treatment <- get_results(fit, coef = "Unstrain",
                              number = nrow(filt_counts), sort = "none")
ma_treatment <- ggplot(data.frame(Amean = fit$Amean, logFC = fit$coef[, "Unstrain"]),
                       aes(x = Amean, y = logFC)) +
  geom_point() +
  labs(x = "Average expression level", y = "Log fold change",
       title = "Treatment effect")
ma_treatment
hist_treatment <- ggplot(results_treatment, aes(x = P.Value)) +
  geom_histogram(binwidth = 0.01) +
  labs(x = "p-value", y = "Number of genes", title = "Treatment effect")
hist_treatment
```

## Use ash for mutliple testing correction

Treatment effect.

```{r ash-treatment}
run_ash <- function(x, coef) {
  # Perform multiple testing correction with adaptive shrinkage (ASH)
  #
  # x - object MArrayLM from eBayes output
  # coef - coefficient tested by eBayes
  stopifnot(class(x) == "MArrayLM", coef %in% colnames(x$coefficients))
  result <- ash(betahat = x$coefficients[, coef],
                sebetahat = x$stdev.unscaled[, coef] * sqrt(x$s2.post),
                df = x$df.total[1])
  return(result)
}

ash_treatment <- run_ash(fit, "Unstrain")
class(ash_treatment)
names(ash_treatment)
sum(ash_treatment$result$svalue < .05)
hist(ash_treatment$result$svalue)
```

# Plots

```{r Plot Functions}
plot_ma <- function(x, qval) {
  # Create MA plot.
  #
  # x - data frame with topTable and ASH output
  #     (columns logFC, AveExpr, and qvalue)
  # qval - qvalue cutoff for calling a gene DE
  #
  stopifnot(is.data.frame(x), c("logFC", "AveExpr", "qvalue") %in% colnames(x),
            is.numeric(qval), qval <= 1, qval >= 0)
  x$highlight <- ifelse(x$qvalue < qval, "darkred", "gray75")
  x$highlight <- factor(x$highlight, levels = c("darkred", "gray75"))
  ggplot(x, aes(x = AveExpr, y = logFC, color = highlight, shape = highlight)) +
    geom_point() +
    labs(x = "Average expression level", y = "Log fold change") +
    scale_color_identity(drop = FALSE) +
    scale_shape_manual(values = c(16, 1), drop = FALSE) +
    theme(legend.position = "none")
#   scale_color_gradient(low = "red", high = "white", limits = c(0, 0.25))
}

plot_volcano <- function(x, qval) {
  # Create volcano plot.
  #
  # x - data frame with topTable and ASH output
  #     (columns logFC, P.Value, and qvalue)
  # qval - qvalue cutoff for calling a gene DE
  #
  stopifnot(is.data.frame(x), c("logFC", "P.Value", "qvalue") %in% colnames(x),
            is.numeric(qval), qval <= 1, qval >= 0)
  x$highlight <- ifelse(x$qvalue < qval, "darkred", "gray75")
  x$highlight <- factor(x$highlight, levels = c("darkred", "gray75"))
  ggplot(x, aes(x = logFC, y = -log10(P.Value), color = highlight)) +
    geom_point(shape = 1) +
    labs(x = "Log fold change",
         y = expression(-log[10] * " p-value")) +
    scale_color_identity(drop = FALSE) +
    theme(legend.position = "none")
}

plot_pval_hist <- function(x, qval) {
  # Create histogram of p-values.
  #
  # x - data frame with topTable and ash output (columns P.Value and qvalue)
  # qval - qvalue cutoff for calling a gene DE
  #
  stopifnot(is.data.frame(x), c("P.Value", "qvalue") %in% colnames(x))
  x$highlight <- ifelse(x$qvalue < qval, "darkred", "gray75")
  x$highlight <- factor(x$highlight, levels = c("darkred", "gray75"))
  ggplot(x, aes(x = P.Value, fill = highlight)) +
    geom_histogram(position = "stack", binwidth = 0.01) +
    scale_fill_identity(drop = FALSE) +
    labs(x = "p-value", y = "Number of genes")
}
```

```{r use Functions}
tests <- colnames(fit$coefficients)
results <- vector(length = length(tests), mode = "list")
names(results) <- tests

for (test in tests[c(1:7)]) {
  # Extract limma results
     print(test)
  results[[test]] <- get_results(fit, coef = test)
  # Add mutliple testing correction with ASH
  output_ash <- run_ash(fit, coef = test)$result
  results[[test]] <- cbind(results[[test]], lfsr = output_ash$lfsr,
                           lfdr = output_ash$lfdr, qvalue = output_ash$qvalue,
                           svalue = output_ash$svalue)
}

#FDR 0.05
plot_ma(results[["Unstrain"]], 0.05)
plot_volcano(results[["Unstrain"]], 0.05)
plot_pval_hist(results[["Unstrain"]], 0.05)
table(results[["Unstrain"]]$qvalue < 0.05)
significant_genes_05 <- row.names(results[["Unstrain"]])[results[["Unstrain"]]$qvalue < 0.05]
significant_symbols_05 <- gene_anno$external_gene_name[match(significant_genes_05, gene_anno$ensembl_gene_id)]
head(significant_symbols_05)
significant_anno_05 <- gene_anno[match(significant_genes_05, gene_anno$ensembl_gene_id),]
```

## Gene ontology analysis with topGO

Use topGO for GO analysis.
It accounts for the nested graph structure of GO terms to prune the number of GO categories tested ([Alexa et al. 2006][Alexa2006]).
Essentially, it decreases the redundancy of the results.

[Alexa2006]: http://www.ncbi.nlm.nih.gov/pubmed/16606683

```{r ks test}
# k-s test making use of ranked based on p-values
genes <- results[["Unstrain"]]$qvalue
names(genes) <- rownames(results[["Unstrain"]])

selection <- function(allScore){ return(allScore < 0.05)} # function that returns TRUE/FALSE for p-values<0.05
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="ensembl")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)

results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment <- goEnrichment %>% 
     mutate(KS = ifelse(grepl("<", KS), 1e-30, KS))
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))
goEnrichment$KS <- as.numeric(goEnrichment$KS)

ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun = mean, position = "dodge") +
    xlab("Biological process") +
    ylab("Enrichment") +
    ggtitle("Title") +
    scale_y_continuous(breaks = round(seq(0, max(-log10(goEnrichment$KS)), by = 2), 1)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

```




# Additional analysis permuting samples:

From Ben Fair: "i have a suggestion for an analysis as a check that all your DE results (like the GO analysis, the enrichment for DE genes from published datasets) are robust… i’m not sure this is statistically perfect but its something that I think makes sense to try, for my own sanity: if you randomly reassign control and treatment labels and re-attampt DE testing, do all of those interesting results go away? the worry i am trying to address, is that you have some enrichment of DE genes in interesting GO categories for example, but is that just because you have more DE power for detecting highly expressed genes in this cell type (even if they are only modestly or not truly DE), and that is what drives the GO enrichment."

```{r permute samples}
#scramble treatment labels (x3) and store the DE genes (each time, randomly assign 9 samples to be strain)
permuted_results <- vector("list", 12)
for(time in 1:6){
     #set up DE and scramble labels
     anno <- pData(RUVsOut)
     x <- paste0(anno$Individual, anno$treatment)

     anno$LibraryPrepBatch <- factor(anno$LibraryPrepBatch, levels = c("1", "2"))
     anno$Replicate <- factor(anno$Replicate, levels = c("1", "2", "3"))

     strain_indices <- sample(nrow(anno), 9)
     anno$treatment <- "Unstrain"
     anno$treatment[strain_indices] <- "Strain"
     design <- model.matrix(~treatment + Individual + W_1 + W_2 + RIN,
                       data = anno)
     colnames(design) <- gsub("treatment", "", colnames(design))

     permuted_results[[time]] <- anno

     #perform DE
     #TMM Normalization
     y <- DGEList(filt_counts)
     y <- calcNormFactors(y, method = "TMM")
     #Voom for differential expression
     v1 <- voom(y, design)
     corfit1 <- duplicateCorrelation(v1, design, block = x)
     corfit1$consensus.correlation
     v2 <- voom(y, design, block = x, correlation = corfit1$consensus.correlation)
     corfit2 <- duplicateCorrelation(v2, design, block = x)
     corfit2$consensus.correlation
     fit <- lmFit(v2, design, block = x,
             correlation = corfit2$consensus.correlation)
     fit <- eBayes(fit)

     tests <- colnames(fit$coefficients)
     results <- vector(length = length(tests), mode = "list")
     names(results) <- tests

     for (test in tests[c(1:7)]) {
     # Extract limma results
      results[[test]] <- get_results(fit, coef = test)
     # Add mutliple testing correction with ASH
     output_ash <- run_ash(fit, coef = test)$result
     results[[test]] <- cbind(results[[test]], lfsr = output_ash$lfsr,
                              lfdr = output_ash$lfdr, qvalue = output_ash$qvalue,
                              svalue = output_ash$svalue)
     }

     #store results
     permuted_results[[6+time]] <- results[["Unstrain"]]
}

#visualize what the permutations look like (what the scrambled labels look like)
for(time in 1:6){
     print(permuted_results[[time]])
}
```

```{r ks test permuted}
for (permutation in 7:12){
     results <- permuted_results[[permutation]]
     
     # k-s test making use of ranked based on p-values
     genes <- results$qvalue
     names(genes) <- rownames(results)

     selection <- function(allScore){ return(allScore < 0.05)} # function that returns TRUE/FALSE for p-values<0.05
     allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="ensembl")
     GOdata <- new("topGOdata",
       ontology="BP",
       allGenes=genes,
       annot=annFUN.GO2genes,
       GO2genes=allGO2genes,
       geneSel=selection,
       nodeSize=10)

     results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
     goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
     goEnrichment <- goEnrichment %>% 
          mutate(KS = ifelse(grepl("<", KS), 1e-30, KS))
     goEnrichment$KS <- as.numeric(goEnrichment$KS)
     goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
     goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
     goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
     goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
     goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
     goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))
     goEnrichment$KS <- as.numeric(goEnrichment$KS)

     plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
         stat_summary(geom = "bar", fun = mean, position = "dodge") +
         xlab("Biological process") +
         ylab("Enrichment") +
         ggtitle("Title") +
         scale_y_continuous(breaks = round(seq(0, max(-log10(goEnrichment$KS)), by = 2), 1)) +
         theme_bw(base_size=24) +
         theme(
             legend.position='none',
             legend.background=element_rect(),
             plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
             axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
             axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
             axis.title=element_text(size=24, face="bold"),
          legend.key=element_blank(),     #removes the border
          legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
          legend.text=element_text(size=18),  #Text size
          title=element_text(size=18)) +
          guides(colour=guide_legend(override.aes=list(size=2.5))) +
          coord_flip()

     print(plot)
}
```


```{r}
for (permutation in 7:12){
     results <- permuted_results[[permutation]]
     
     # k-s test making use of ranked based on p-values
     genes <- results$qvalue
     names(genes) <- rownames(results)

     selection <- function(allScore){ return(allScore < 0.05)} # function that returns TRUE/FALSE for p-values<0.05
     allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="ensembl")
     GOdata <- new("topGOdata",
       ontology="BP",
       allGenes=genes,
       annot=annFUN.GO2genes,
       GO2genes=allGO2genes,
       geneSel=selection,
       nodeSize=10)

     results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
     goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
     goEnrichment <- goEnrichment %>% 
          mutate(KS = ifelse(grepl("<", KS), 1e-30, KS))
     goEnrichment$KS <- as.numeric(goEnrichment$KS)
     goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
     goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
     goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
     goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
     goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
     goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))
     goEnrichment$KS <- as.numeric(goEnrichment$KS)

     print(goEnrichment$Term)
}
```


# Look at enrichment of DE genes amongst previously published datasets of OA related genes

```{r load}
library(dplyr)
# load gene annotations
gene_anno <- read.delim("data/gene-annotation.txt",
                        sep = "\t")

DE_results <- results
threshold <- 0.05
DE_genes_ensg <- rownames(DE_results[["Unstrain"]])[DE_results[["Unstrain"]]$qvalue < threshold]
DE_genes <- gene_anno$external_gene_name[match(DE_genes_ensg, gene_anno$ensembl_gene_id)]
other_genes_ensg <- rownames(DE_results[["Unstrain"]])[DE_results[["Unstrain"]]$qvalue >= threshold]
other_genes <- gene_anno$external_gene_name[match(other_genes_ensg, gene_anno$ensembl_gene_id)]

#Genes from https://www.nature.com/articles/s41588-018-0327-1
#Table S4: Functional genomics annotations for all genes within 500kb of the osteoarthritis-associated variants. (subset to those with any evidence)
fungenomic <- c("ENSG00000060718", "ENSG00000270962", "ENSG00000150337", "ENSG00000264522", "ENSG00000136631", "ENSG00000023902", "ENSG00000117362", "ENSG00000159208", "ENSG00000163125", "ENSG00000143374", "ENSG00000143369", "ENSG00000143420", "ENSG00000163131", "ENSG00000143387", "ENSG00000143412", "ENSG00000143409", "ENSG00000213190", "ENSG00000120334", "ENSG00000117593", "ENSG00000135870", "ENSG00000227373", "ENSG00000152061", "ENSG00000135862", "ENSG00000058085", "ENSG00000116701", "ENSG00000198756", "ENSG00000198860", "ENSG00000143353", "ENSG00000228536", "ENSG00000116141", "ENSG00000203667", "ENSG00000185420", "ENSG00000018699", "ENSG00000049323", "ENSG00000119812", "ENSG00000059728", "ENSG00000244617", "ENSG00000116001", "ENSG00000116005", "ENSG00000143977", "ENSG00000035141", "ENSG00000163235", "ENSG00000116039", "ENSG00000124357", "ENSG00000135636", "ENSG00000115419", "ENSG00000115415", "ENSG00000138378", "ENSG00000204217", "ENSG00000138439", "ENSG00000163596", "ENSG00000138380", "ENSG00000138442", "ENSG00000173166", "ENSG00000163599", "ENSG00000178467", "ENSG00000178252", "ENSG00000178035", "ENSG00000172037", "ENSG00000145020", "ENSG00000145029", "ENSG00000173402", "ENSG00000164068", "ENSG00000173540", "ENSG00000176095", "ENSG00000185614", "ENSG00000182179", "ENSG00000164076", "ENSG00000234667", "ENSG00000164078", "ENSG00000228008", "ENSG00000004534", "ENSG00000003756", "ENSG00000001617", "ENSG00000114353", "ENSG00000214706", "ENSG00000114378", "ENSG00000068001", "ENSG00000126062", "ENSG00000007402", "ENSG00000114737", "ENSG00000114767", "ENSG00000090097", "ENSG00000248487", "ENSG00000114779", "ENSG00000114786", "ENSG00000162244", "ENSG00000164087", "ENSG00000247596", "ENSG00000164088", "ENSG00000164091", "ENSG00000168237", "ENSG00000239557", "ENSG00000010318", "ENSG00000010322", "ENSG00000168268", "ENSG00000163939", "ENSG00000163938", "ENSG00000016864", "ENSG00000114902", "ENSG00000270941", "ENSG00000114904", "ENSG00000055957", "ENSG00000162267", "ENSG00000055955", "ENSG00000272573", "ENSG00000163935", "ENSG00000163932", "ENSG00000163931", "ENSG00000127415", "ENSG00000127418", "ENSG00000178222", "ENSG00000159692", "ENSG00000163945", "ENSG00000244459", "ENSG00000174137", "ENSG00000163950", "ENSG00000270195", "ENSG00000168936", "ENSG00000013810", "ENSG00000068078", "ENSG00000168924", "ENSG00000109685", "ENSG00000130997", "ENSG00000159733", "ENSG00000157869", "ENSG00000109705", "ENSG00000109606", "ENSG00000248444", "ENSG00000109610", "ENSG00000181982", "ENSG00000168228", "ENSG00000053900", "ENSG00000091490", "ENSG00000138814", "ENSG00000138821", "ENSG00000109320", "ENSG00000109332", "ENSG00000145354", "ENSG00000164039", "ENSG00000253558", "ENSG00000132842", "ENSG00000113273", "ENSG00000132840", "ENSG00000145692", "ENSG00000132837", "ENSG00000156427", "ENSG00000174705", "ENSG00000079691", "ENSG00000272462", "ENSG00000272810", "ENSG00000187837", "ENSG00000010704", "ENSG00000218281", "ENSG00000124508", "ENSG00000111801", "ENSG00000112763", "ENSG00000124635", "ENSG00000196787", "ENSG00000197903", "ENSG00000204305", "ENSG00000204304", "ENSG00000213654", "ENSG00000198502", "ENSG00000224796", "ENSG00000223865", "ENSG00000204248", "ENSG00000204227", "ENSG00000030110", "ENSG00000096433", "ENSG00000161896", "ENSG00000161904", "ENSG00000137207", "ENSG00000171453", "ENSG00000112715", "ENSG00000181577", "ENSG00000137216", "ENSG00000112759", "ENSG00000146232", "ENSG00000196284", "ENSG00000271857", "ENSG00000124813", "ENSG00000196284")
fungenomic1 <- c("ENSG00000124813", "ENSG00000271857", "ENSG00000112782", "ENSG00000172348", "ENSG00000112175", "ENSG00000111799", "ENSG00000112697", "ENSG00000118407", "ENSG00000196586", "ENSG00000105852", "ENSG00000004799", "ENSG00000158560", "ENSG00000004864", "ENSG00000006377", "ENSG00000105880", "ENSG00000253981", "ENSG00000173295", "ENSG00000104626", "ENSG00000173281", "ENSG00000248538", "ENSG00000175806", "ENSG00000147697", "ENSG00000153310", "ENSG00000106688", "ENSG00000096968", "ENSG00000136867", "ENSG00000119321", "ENSG00000176386", "ENSG00000148229", "ENSG00000157653", "ENSG00000157657", "ENSG00000196739", "ENSG00000229314", "ENSG00000228278", "ENSG00000181634", "ENSG00000106952", "ENSG00000041982", "ENSG00000196739", "ENSG00000229314", "ENSG00000228278", "ENSG00000181634", "ENSG00000106952", "ENSG00000041982", "ENSG00000182752", "ENSG00000136869", "ENSG00000167081", "ENSG00000136944", "ENSG00000169155", "ENSG00000136859", "ENSG00000136830", "ENSG00000066382", "ENSG00000110047", "ENSG00000068971", "ENSG00000110025", "ENSG00000168060", "ENSG00000149809", "ENSG00000162298", "ENSG00000014216", "ENSG00000162241", "ENSG00000126391", "ENSG00000245532", "ENSG00000251562", "ENSG00000142186", "ENSG00000168056", "ENSG00000260233", "ENSG00000176973", "ENSG00000173338", "ENSG00000173327", "ENSG00000173039", "ENSG00000172922", "ENSG00000172638", "ENSG00000175592", "ENSG00000175550", "ENSG00000175334", "ENSG00000179292", "ENSG00000174684", "ENSG00000174576", "ENSG00000174547", "ENSG00000174516", "ENSG00000174165", "ENSG00000172638", "ENSG00000175592", "ENSG00000175550", "ENSG00000175334", "ENSG00000179292", "ENSG00000174684", "ENSG00000174576", "ENSG00000174547", "ENSG00000174516", "ENSG00000174165", "ENSG00000173914", "ENSG00000173898", "ENSG00000173599", "ENSG00000173237", "ENSG00000173156", "ENSG00000172932", "ENSG00000172613", "ENSG00000175514", "ENSG00000167800", "ENSG00000137492", "ENSG00000204529", "ENSG00000254632", "ENSG00000255100", "ENSG00000182704", "ENSG00000261578", "ENSG00000078124", "ENSG00000149269", "ENSG00000227376", "ENSG00000048649", "ENSG00000064102", "ENSG00000111790", "ENSG00000110841", "ENSG00000174236", "ENSG00000230519", "ENSG00000061794", "ENSG00000087494", "ENSG00000139263", "ENSG00000118596", "ENSG00000237176", "ENSG00000111581", "ENSG00000175782", "ENSG00000135679", "ENSG00000111605", "ENSG00000090382", "ENSG00000166225", "ENSG00000166226", "ENSG00000127328", "ENSG00000166268", "ENSG00000139318", "ENSG00000070961", "ENSG00000177889", "ENSG00000120833", "ENSG00000258274", "ENSG00000169372", "ENSG00000173588", "ENSG00000089041", "ENSG00000135124", "ENSG00000188735", "ENSG00000256546", "ENSG00000184047", "ENSG00000284934", "ENSG00000256304", "ENSG00000139726", "ENSG00000130787", "ENSG00000139722", "ENSG00000090975", "ENSG00000139726", "ENSG00000130787", "ENSG00000139722", "ENSG00000150967", "ENSG00000090975", "ENSG00000051825", "ENSG00000130921", "ENSG00000235423", "ENSG00000111328", "ENSG00000139697", "ENSG00000150977")
fungenomic2 <- c("ENSG00000184209", "ENSG00000111364", "ENSG00000250091", "ENSG00000178882", "ENSG00000196498", "ENSG00000140285", "ENSG00000138592", "ENSG00000242737", "ENSG00000259618", "ENSG00000081014", "ENSG00000183578", "ENSG00000137869", "ENSG00000128849", "ENSG00000128918", "ENSG00000137845", "ENSG00000169032", "ENSG00000188501", "ENSG00000166949", "ENSG00000033800", "ENSG00000129038", "ENSG00000167178", "ENSG00000129009", "ENSG00000137868", "ENSG00000140481", "ENSG00000138623", "ENSG00000179151", "ENSG00000140465", "ENSG00000103653", "ENSG00000140506", "ENSG00000140474", "ENSG00000140497", "ENSG00000178802", "ENSG00000178741", "ENSG00000178718", "ENSG00000140398", "ENSG00000140400", "ENSG00000169410", "ENSG00000177971", "ENSG00000173548", "ENSG00000173546", "ENSG00000277639", "ENSG00000103479", "ENSG00000166971", "ENSG00000103494", "ENSG00000140718", "ENSG00000177508", "ENSG00000062038", "ENSG00000103047", "ENSG00000258429", "ENSG00000102908", "ENSG00000198373", "ENSG00000157322", "ENSG00000255185", "ENSG00000157353", "ENSG00000103051", "ENSG00000189091", "ENSG00000157368", "ENSG00000051523", "ENSG00000167513", "ENSG00000141012", "ENSG00000167522", "ENSG00000197912", "ENSG00000015413", "ENSG00000131165", "ENSG00000158792", "ENSG00000158805", "ENSG00000187741", "ENSG00000258839", "ENSG00000258947", "ENSG00000223959", "ENSG00000197879", "ENSG00000132386", "ENSG00000132383", "ENSG00000108963", "ENSG00000177374", "ENSG00000070366", "ENSG00000262333", "ENSG00000167721", "ENSG00000070444", "ENSG00000127804", "ENSG00000108578", "ENSG00000263603", "ENSG00000266490", "ENSG00000181481", "ENSG00000265798", "ENSG00000196712", "ENSG00000126861", "ENSG00000185862", "ENSG00000126860", "ENSG00000131242", "ENSG00000108651", "ENSG00000178691", "ENSG00000126858", "ENSG00000172992", "ENSG00000184922", "ENSG00000159314", "ENSG00000225190", "ENSG00000266918", "ENSG00000214425", "ENSG00000264070", "ENSG00000204652", "ENSG00000120088", "ENSG00000185294", "ENSG00000186868", "ENSG00000262539", "ENSG00000261575", "ENSG00000266497", "ENSG00000238083", "ENSG00000185829", "ENSG00000073969", "ENSG00000108379", "ENSG00000108433", "ENSG00000172992", "ENSG00000136448", "ENSG00000184922", "ENSG00000225190", "ENSG00000108433", "ENSG00000141376", "ENSG00000121068", "ENSG00000121075", "ENSG00000256124", "ENSG00000125398", "ENSG00000134490", "ENSG00000101782", "ENSG00000141458", "ENSG00000053747", "ENSG00000141447", "ENSG00000105088", "ENSG00000080573", "ENSG00000130812", "ENSG00000130810", "ENSG00000244165", "ENSG00000130816", "ENSG00000267534", "ENSG00000090339", "ENSG00000105376", "ENSG00000220201", "ENSG00000267303", "ENSG00000105397", "ENSG00000065989", "ENSG00000129353", "ENSG00000267100", "ENSG00000129351", "ENSG00000079805", "ENSG00000099203", "ENSG00000127616", "ENSG00000197256", "ENSG00000105518", "ENSG00000205517", "ENSG00000198003", "ENSG00000130176", "ENSG00000102575", "ENSG00000105223", "ENSG00000268366", "ENSG00000090013", "ENSG00000090006", "ENSG00000105245", "ENSG00000086544", "ENSG00000077312", "ENSG00000261857", "ENSG00000233622", "ENSG00000167601", "ENSG00000105329", "ENSG00000105388", "ENSG00000160570", "ENSG00000105723", "ENSG00000105722", "ENSG00000079462", "ENSG00000167619", "ENSG00000105429", "ENSG00000167646", "ENSG00000080031", "ENSG00000160471", "ENSG00000095752", "ENSG00000187902", "ENSG00000090971", "ENSG00000179954", "ENSG00000171443", "ENSG00000131848", "ENSG00000078747", "ENSG00000101460", "ENSG00000101464", "ENSG00000131069", "ENSG00000078814", "ENSG00000100991", "ENSG00000088298", "ENSG00000101000", "ENSG00000242372", "ENSG00000101019", "ENSG00000125965", "ENSG00000126001", "ENSG00000125991", "ENSG00000088340", "ENSG00000244462", "ENSG00000025293", "ENSG00000171222", "ENSG00000231123", "ENSG00000157551", "ENSG00000157554", "ENSG00000183527", "ENSG00000205581", "ENSG00000189306", "ENSG00000182841", "ENSG00000100243", "ENSG00000128274", "ENSG00000274717", "ENSG00000100266", "ENSG00000100294", "ENSG00000100300", "ENSG00000100304", "ENSG00000159307", "ENSG00000100347", "ENSG00000188677")
fungenomic <- c(fungenomic, fungenomic1, fungenomic2)



#doi: https://doi.org/10.1101/835850
#Supplementary Table 2. Genes with significant cross-omics differences between high-grade and low-grade cartilage. (transcriptomic, proteomic both were significant)
s2 <- c("ENSG00000164163", "ENSG00000100997", "ENSG00000154175", "ENSG00000131473", "ENSG00000068366", "ENSG00000133627", "ENSG00000148700", "ENSG00000144843", "ENSG00000035687", "ENSG00000106624", "ENSG00000173744", "ENSG00000177674", "ENSG00000101444", "ENSG00000004455", "ENSG00000085662", "ENSG00000187134", "ENSG00000151632", "ENSG00000059573", "ENSG00000128918", "ENSG00000184254", "ENSG00000116337", "ENSG00000196510", "ENSG00000151150", "ENSG00000166825", "ENSG00000189058", "ENSG00000142192", "ENSG00000140750", "ENSG00000135931", "ENSG00000100299", "ENSG00000104763", "ENSG00000160072", "ENSG00000071553", "ENSG00000006453", "ENSG00000107949", "ENSG00000099968", "ENSG00000104164", "ENSG00000183336", "ENSG00000113460", "ENSG00000186665", "ENSG00000106392", "ENSG00000223953", "ENSG00000159403", "ENSG00000123838", "ENSG00000169239", "ENSG00000042493", "ENSG00000105519", "ENSG00000153048", "ENSG00000164305", "ENSG00000196954", "ENSG00000007080", "ENSG00000152492", "ENSG00000055813", "ENSG00000110104", "ENSG00000115009", "ENSG00000260916", "ENSG00000000971", "ENSG00000016391", "ENSG00000255112", "ENSG00000130724", "ENSG00000054938", "ENSG00000163815", "ENSG00000169583", "ENSG00000159212", "ENSG00000111726", "ENSG00000164237", "ENSG00000064666", "ENSG00000117519", "ENSG00000168434", "ENSG00000204291", "ENSG00000182871", "ENSG00000164692", "ENSG00000168542", "ENSG00000187498", "ENSG00000134871", "ENSG00000130635", "ENSG00000112280", "ENSG00000184432", "ENSG00000181789", "ENSG00000160917", "ENSG00000110090", "ENSG00000157184", "ENSG00000121005", "ENSG00000006016", "ENSG00000095713", "ENSG00000160213", "ENSG00000159692", "ENSG00000117151", "ENSG00000064601", "ENSG00000117984", "ENSG00000083799", "ENSG00000138061", "ENSG00000135929", "ENSG00000110063", "ENSG00000153904", "ENSG00000162733", "ENSG00000198171", "ENSG00000168872", "ENSG00000165732", "ENSG00000145833", "ENSG00000213782", "ENSG00000023697", "ENSG00000105928", "ENSG00000065357", "ENSG00000102967", "ENSG00000105821", "ENSG00000197635") 
s2.1<- c("ENSG00000176978", "ENSG00000188641", "ENSG00000134755", "ENSG00000103356", "ENSG00000115380", "ENSG00000113790", "ENSG00000115211", "ENSG00000145191", "ENSG00000155849", "ENSG00000074800", "ENSG00000159023", "ENSG00000082397", "ENSG00000177106", "ENSG00000157554", "ENSG00000023318", "ENSG00000081177", "ENSG00000178896", "ENSG00000120699", "ENSG00000164687", "ENSG00000119812", "ENSG00000078098", "ENSG00000116120", "ENSG00000077942", "ENSG00000079459", "ENSG00000160752", "ENSG00000154783", "ENSG00000113578", "ENSG00000132589", "ENSG00000115414", "ENSG00000162998", "ENSG00000134363", "ENSG00000163430", "ENSG00000087086", "ENSG00000179271", "ENSG00000117308", "ENSG00000164574", "ENSG00000131386", "ENSG00000100626", "ENSG00000023909", "ENSG00000131459", "ENSG00000006625", "ENSG00000137563", "ENSG00000170266", "ENSG00000124767", "ENSG00000173221", "ENSG00000108010", "ENSG00000182512", "ENSG00000163938", "ENSG00000113552", "ENSG00000100522", "ENSG00000135677", "ENSG00000171723", "ENSG00000166123", "ENSG00000164294", "ENSG00000030582", "ENSG00000105447", "ENSG00000082701", "ENSG00000100577", "ENSG00000188342", "ENSG00000144366", "ENSG00000206172", "ENSG00000244734", "ENSG00000111911", "ENSG00000170144", "ENSG00000138668", "ENSG00000164070", "ENSG00000152137", "ENSG00000166033", "ENSG00000170801", "ENSG00000196305", "ENSG00000163453", "ENSG00000172349", "ENSG00000172458", "ENSG00000106348", "ENSG00000104613", "ENSG00000198001", "ENSG00000169047", "ENSG00000172183", "ENSG00000136156", "ENSG00000205726", "ENSG00000197256", "ENSG00000102781", "ENSG00000156113", "ENSG00000116685", "ENSG00000102753", "ENSG00000135862", "ENSG00000185896", "ENSG00000111716", "ENSG00000166816", "ENSG00000116977", "ENSG00000005156", "ENSG00000050405", "ENSG00000176619", "ENSG00000161654", "ENSG00000130332", "ENSG00000119681", "ENSG00000168056", "ENSG00000139329", "ENSG00000011009", "ENSG00000166963", "ENSG00000131711", "ENSG00000108984", "ENSG00000277443", "ENSG00000124159", "ENSG00000185432", "ENSG00000166482", "ENSG00000131446", "ENSG00000008394", "ENSG00000123342", "ENSG00000107186", "ENSG00000105926", "ENSG00000133030", "ENSG00000011028", "ENSG00000125901", "ENSG00000113048", "ENSG00000053372", "ENSG00000099810", "ENSG00000150712", "ENSG00000132382", "ENSG00000105357", "ENSG00000196586", "ENSG00000138119", "ENSG00000105835", "ENSG00000095380", "ENSG00000124479", "ENSG00000104419", "ENSG00000166136", "ENSG00000049759", "ENSG00000162599", "ENSG00000239672")
s2.2 <- c("ENSG00000243678", "ENSG00000123609", "ENSG00000152465", "ENSG00000055044", "ENSG00000139910", "ENSG00000119655", "ENSG00000215440", "ENSG00000182446", "ENSG00000159899", "ENSG00000181019", "ENSG00000184162", "ENSG00000148572", "ENSG00000065320", "ENSG00000090273", "ENSG00000136159", "ENSG00000132661", "ENSG00000136811", "ENSG00000127083", "ENSG00000178814", "ENSG00000141447", "ENSG00000070882", "ENSG00000117859", "ENSG00000149380", "ENSG00000170515", "ENSG00000111845", "ENSG00000163346", "ENSG00000183570", "ENSG00000148843", "ENSG00000067992", "ENSG00000107438", "ENSG00000131435", "ENSG00000139197", "ENSG00000178921", "ENSG00000141959", "ENSG00000247077", "ENSG00000101856", "ENSG00000067177", "ENSG00000139289", "ENSG00000173868", "ENSG00000153823", "ENSG00000087842", "ENSG00000123143", "ENSG00000188257", "ENSG00000124181", "ENSG00000198805", "ENSG00000102978", "ENSG00000105854", "ENSG00000133110", "ENSG00000131626", "ENSG00000196262", "ENSG00000066027", "ENSG00000188783", "ENSG00000126457", "ENSG00000100462", "ENSG00000136875", "ENSG00000147224", "ENSG00000157778", "ENSG00000117569", "ENSG00000148344", "ENSG00000134247", "ENSG00000124212", "ENSG00000196396", "ENSG00000132670", "ENSG00000173482", "ENSG00000105426", "ENSG00000183255", "ENSG00000177192", "ENSG00000089159", "ENSG00000183010", "ENSG00000100504", "ENSG00000116260", "ENSG00000112210", "ENSG00000132341", "ENSG00000099901", "ENSG00000106538", "ENSG00000113643", "ENSG00000159200", "ENSG00000133119", "ENSG00000155366", "ENSG00000177189", "ENSG00000116954", "ENSG00000133818", "ENSG00000114767", "ENSG00000147655", "ENSG00000133318", "ENSG00000196154", "ENSG00000177409", "ENSG00000184178", "ENSG00000006747", "ENSG00000132581", "ENSG00000117118", "ENSG00000100445", "ENSG00000157020", "ENSG00000138674", "ENSG00000075223", "ENSG00000138758", "ENSG00000166401", "ENSG00000135919", "ENSG00000109686", "ENSG00000147010", "ENSG00000174705", "ENSG00000104635", "ENSG00000102172", "ENSG00000135632", "ENSG00000145335", "ENSG00000077312", "ENSG00000172164", "ENSG00000169371", "ENSG00000124104", "ENSG00000089006", "ENSG00000095637", "ENSG00000128487", "ENSG00000115306", "ENSG00000075142", "ENSG00000100883", "ENSG00000102359", "ENSG00000111786", "ENSG00000114850", "ENSG00000118804", "ENSG00000101109", "ENSG00000165283", "ENSG00000023734", "ENSG00000125755", "ENSG00000159082", "ENSG00000136111", "ENSG00000106638", "ENSG00000184905", "ENSG00000204065", "ENSG00000204071", "ENSG00000120708", "ENSG00000169231", "ENSG00000051596", "ENSG00000100296", "ENSG00000134809", "ENSG00000105197", "ENSG00000035862", "ENSG00000142910", "ENSG00000137462", "ENSG00000125304", "ENSG00000135048", "ENSG00000136842", "ENSG00000138594", "ENSG00000123610", "ENSG00000078902", "ENSG00000025772", "ENSG00000067369", "ENSG00000136527", "ENSG00000135148", "ENSG00000126602", "ENSG00000132481", "ENSG00000162222", "ENSG00000123416", "ENSG00000183785", "ENSG00000025708", "ENSG00000154277", "ENSG00000118939", "ENSG00000132478", "ENSG00000183696", "ENSG00000090686", "ENSG00000160293", "ENSG00000162692", "ENSG00000213585", "ENSG00000136100", "ENSG00000139722", "ENSG00000138442", "ENSG00000065183", "ENSG00000196363", "ENSG00000196419", "ENSG00000139131", "ENSG00000065978", "ENSG00000140836", "ENSG00000100711", "ENSG00000109917")
cross_omic <- c(s2, s2.1, s2.2)
```

# Enrichment Function

Use fisher's exact test to compute significant enrichment.

```{r function}
enrich <- function(genes_of_interest, DEgenes, backgroundgenes){
     #fill contingency table
     DE_interest <- length(intersect(genes_of_interest, DEgenes))
     background_interest <- length(intersect(genes_of_interest, backgroundgenes))
     DE_non_interest <- length(DEgenes) - DE_interest
     background_non_interest <- length(backgroundgenes) - background_interest
     contingency_table <- matrix(c(DE_interest, background_interest, DE_non_interest, background_non_interest), nrow = 2, ncol = 2, byrow = TRUE)
     
     #perform test
     print(contingency_table)
     return(fisher.test(contingency_table, alternative="greater"))
}


enrich(fungenomic, DE_genes_ensg, other_genes_ensg)
enrich(cross_omic, DE_genes_ensg, other_genes_ensg)
```

# Look at enrichments from permuted DE analysis (scrambled strain labels to make sure the FET results aren't just because some genes are more likely to be detected as DE in these cells)

```{r permuted}
threshold <- 0.05
for(permutation in 7:12){
     print(permutation)
     DE_results <- permuted_results[[permutation]]
     DE_genes_ensg <- rownames(DE_results)[DE_results$qvalue < threshold]
     DE_genes <- gene_anno$external_gene_name[match(DE_genes_ensg, gene_anno$ensembl_gene_id)]
     other_genes_ensg <- rownames(DE_results)[DE_results$qvalue >= threshold]
     other_genes <- gene_anno$external_gene_name[match(other_genes_ensg, gene_anno$ensembl_gene_id)]

     print(enrich(fungenomic, DE_genes_ensg, other_genes_ensg))
     print(enrich(cross_omic, DE_genes_ensg, other_genes_ensg))
}
```

